03_ocp_repo_sync.sh

```diff
@@ -12,3 +12,11 @@ export REPO_PATH="$GOPATH/src"
 
 # Install baremetal-operator
 sync_repo_and_patch github.com/metal3-io/baremetal-operator https://github.com/metal3-io/baremetal-operator.git
+
+# Rebase machine-api-operator if it is present
+MAO_DIR="${GOPATH}/src/github.com/openshift/machine-api-operator"
+if [ -d "${MAO_DIR}" ]; then
+    pushd ${MAO_DIR}
+    git pull --rebase
+    popd
+fi
```

Comment:
Could we remove this? IMHO it's better to let a developer set the repo up the way the want to be consistent with how we handle other git checkouts (e.g. the installer). Also, if they pulled a patch from github and did `git apply` or manually made some changes to test, I think git will error out here as well as the rebase would overwrite unstaged changes.

Reply:
Perhaps I'm being naive, but I'd expect a script named repo_sync to... sync the repos? Devs always have the option of not running it. And this is consistent, at least in spirit, with what we do with the baremetal operator on line 14.

Reply:
If I'm testing something I want to set up the repo exactly as I expect.baremetal-operator is different as we always sync it and it's required to have it.

The installer is more analogous to me, as you only have it checked out if you're doing development, and if the directory exists we leave it alone, otherwise we clone it fresh. If I'm doing development, I want the repo left as the way I have it. 

@sadasu also mentioned above she would prefer a way to turn this feature off.